src/app/qgisapp.cpp

```diff
@@ -7451,23 +7451,23 @@ void QgisApp::setupLayoutManagerConnections()
 
 void QgisApp::setupDuplicateFeaturesAction()
 {
-  QgsMapLayerAction *action = new QgsMapLayerAction( QString( tr( "Duplicate feature" ) ),
-      this, QgsMapLayerAction::AllActions,
-      QgsApplication::getThemeIcon( QStringLiteral( "/mActionDuplicateFeature.svg" ) ) );
+  mDuplicateFeatureAction.reset( new QgsMapLayerAction( QString( tr( "Duplicate feature" ) ),
```

Comment:
I think you'll need to remove the parents here... Otherwise they have dual ownership

Reply:
Also no reason to wrap the return value of `tr` in yet another `QString` constructor I guess.